cmd/sync/sync.go

```diff
@@ -146,17 +133,36 @@ func (app *syncApp) start(cmd *cobra.Command, args []string) {
 	}
 
 	txpool := state.NewTxMemPool()
-	atxpool := activation.NewAtxMemPool()
 
-	app.sync = sync.NewSyncWithMocks(atxdbStore, mshdb, txpool, atxpool, swarm, poetDb, conf, goldenATXID, types.LayerID(expectedLayers), poetDbStore)
+	app.logger = log.NewDefault("sync_test")
+	app.logger.Info("new sync tester")
+
+	layersPerEpoch := app.Config.LayersPerEpoch
+	atxdb := activation.NewDB(atxdbStore, &mockIStore{}, mshdb, uint16(layersPerEpoch), goldenATXID, &validatorMock{}, lg.WithOptions(log.Nop))
+	dbs := &allDbs{
+		atxdb:       atxdb,
+		atxdbStore:  atxdbStore,
+		poetDb:      poetDb,
+		poetStorage: poetDbStore,
+		mshdb:       mshdb,
+	}
+	msh := createMockMesh(dbs, txpool, app.logger)
+	app.msh = msh
+	layerFetch := createMockFetcher(dbs, msh, swarm, app.logger)
```

Comment:
The purpose of the sync test is to test actual syncing of data on a live network so I think we should use actual fetcher instead

Reply:
this is a real Fetcher with a mockTxProcessor. do you want me to rename the method instead?

Reply:
offline conversation. resolved.